src/consul.rs

```diff
@@ -490,5 +547,14 @@ async fn ensure_config_exists(
         index.replace(new_index);
     };
 
-    Ok(())
+    let res = res
+        .json::<Vec<KVResponse>>()
+        .await
+        .context("Failed to parse KV response")?;
+
+    ensure!(res.len() == 1, "Expected 1 KV response, got {}", res.len());
```

Comment:
When would that be the case? Add a comment or improve the message to help someone debug this if this is important to enforce.